CrowServe/CrownLinkProtocol.h

```diff
@@ -79,17 +100,81 @@ struct AdvertisementsRequest : Header {
 };
 
 struct AdvertisementsResponse : Header {
-    std::vector<AdFile> ads;
+    std::vector<AdFile> ad_files;
 };
+NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(AdvertisementsResponse, ad_files)
 
 struct EchoRequest : Header {
     std::string message;
 };
+NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(EchoRequest, message)
 
-struct EchoResponse {
-    Header header;
-    
+struct EchoResponse : Header {
     std::string message;
 };
+NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(EchoResponse, message)
+
+class CrownLinkProtocol {
+public:
+    template <typename Handler>
+    void handle(const MessageType message_type, const std::span<char> message, const Handler& handler) {
```

Comment:
const goes inside of span, outside it doesn't have any practical effect.
`std::span<const char>`